api/server/router/system/backend.go

```diff
@@ -0,0 +1,18 @@
+package system
+
+import (
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/cliconfig"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/docker/docker/pkg/parsers/filters"
+)
+
+// Backend is the methods that need to be implemented to provide
+// system specific functionality.
+type Backend interface {
+	SystemInfo() (*types.Info, error)
+	SystemVersion() types.Version
+	SubscribeToEvents(since, sinceNano int64, ef filters.Args) ([]*jsonmessage.JSONMessage, chan interface{})
+	UnsubscribeFromEvents(chan interface{})
+	AuthenticateToRegistry(authConfig *cliconfig.AuthConfig) (string, error)
```

Comment:
Yes, that's the idea, I still don't know how completely true it's going to be, but we can always iterate.

For instance, a first use case is that swarm should be able to reuse the api endpoints and write their own backends.